packages/office-ui-fabric-react/src/components/pickers/TagPicker/TagItem.scss

```diff
@@ -38,7 +38,6 @@
   overflow: hidden;
   text-overflow: ellipsis;
   white-space: nowrap;
-  min-width: 100px;
```

Comment:
You need to have some kind of min-width here. Items in a flex container only shrink if a min-width is specified. So even if that min-width is 0 it still needs it set.